### DISCUSSION TITLE
Create folders for pre-made data types

### DISCUSSION THREAD

**Discussion by simonbusborg** [umbraco/Umbraco-CMS#10714] [Features and ideas]
Create folders for pre-made data types. Add all pre-made data types to a folder called for example "System", then divide them into subfolders "Media", "Picker", "True/False" etc.

All newly created data types will be added to the root of Data types.

**Comment by nul800sebastiaan** [comment]
I'll just link to a previous discussion around the same topic (haven't re-read it yet, so not sure what the conclusion was there) https://github.com/umbraco/Umbraco-CMS/issues/8729

**Reply by simonbusborg** [reply]
Not quite the same issue/feature as in the other thread, as it's more hands down to the type of predefined data types, and not how you present/structure the presentation around them :)

**Comment by bjarnef** [comment]
I have thought of something similar as Umbraco by default comes with many datatype instances, but not sure what would be the best way to group these into folders. It could affect new installs, but not existing installs.

Grid, List View and Block List editor can be used as only a few datatype instances or many instances. I tend to group the Block List editor instances in a folder in working projects.

Sometimes you may also prefer to group "pickers" in folders or checkbox/radiobutton/dropdown lists, but it probably depends on the specific project and personal preferences.

However I think it could make sense to group the datatype instances by default in folders, while developers still have the option to move datatypes to different folders.

Furthermore we now also have several Upload datatype instances.

![image](https://user-images.githubusercontent.com/2919859/126907778-9ab4a8b4-e3ff-442e-b517-954573ee3616.png)

![image](https://user-images.githubusercontent.com/2919859/126907796-4ecafff3-bcb4-4c1c-94a4-99ac5653b8da.png)
